pkg/types/validation/installconfig.go

```diff
@@ -1117,3 +1099,47 @@ func validateCustomFeatureGates(c *types.InstallConfig) field.ErrorList {
 
 	return allErrs
 }
+
+// validateGatedFeatures ensures that any gated features used in
+// the install config are enabled.
+func validateGatedFeatures(c *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// legacy feature gating
+	if c.FeatureSet != configv1.TechPreviewNoUpgrade {
+		errMsg := "the TechPreviewNoUpgrade feature set must be enabled to use this field"
+
+		if c.OpenStack != nil {
+			for _, f := range openstackvalidation.FilledInTechPreviewFields(c) {
+				allErrs = append(allErrs, field.Forbidden(f, errMsg))
+			}
+		}
+	}
```

Comment:
Not updating this because it will be removed by #7570, which should merge soon.

Reply:
@MaysaMacedo @EmilienM @pierreprinetti @mandre PTAL for future feature gating needs